sites/svelte-5-preview/src/routes/docs/content/01-api/04-event-handlers.md

```diff
@@ -0,0 +1,180 @@
+	let { children, ...props } = $props();
+</script>
+
+<button {...props}>
+	{@render children()}
+</button>
+```
+
+## Event modifiers
+
+In Svelte 4, you can add event modifiers to handlers:
+
+```svelte
+<button on:click|once|preventDefault={handler}>...</button>
+```
+
+Modifiers are specific to `on:` and as such do not work with modern event handlers. Adding things like `event.preventDefault()` inside the handler itself is preferable, since all the logic lives in one place rather than being split between handler and modifiers.
+
+Since event handlers are just functions, you can create your own wrappers as necessary:
+
+```svelte
+<script>
+	function once(fn) {
+		return function (event) {
+			if (fn) fn.call(this, event);
+			fn = null;
+		};
+	}
+
+	function preventDefault(fn) {
+		return function (event) {
+			event.preventDefault();
+			fn.call(this, event);
+		};
+	}
+</script>
+
+<button onclick={once(preventDefault(handler))}>...</button>
+```
+
+There are three modifiers — `capture`, `passive` and `nonpassive` — that can't be expressed as wrapper functions, since they need to be applied when the event handler is bound rather than when it runs.
+
+For `capture`, we add the modifier to the event name:
+
+```svelte
+<button onclickcapture={...}>...</button>
+```
+
+Changing the [`passive`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#using_passive_listeners) option of an event handler, meanwhile, is not something to be done lightly. If you have a use case for it — and you probably don't! - then you will need to use an action to apply the event handler yourself.
+
+## Multiple event handlers
+
+In Svelte 4, this is possible:
+
+```svelte
+<button on:click={one} on:click={two}>...</button>
+```
+
+This is something of an anti-pattern, since it impedes readability (if there are many attributes, it becomes harder to spot that there are two handlers unless they are right next to each other) and implies that the two handlers are independent, when in fact something like `event.stopImmediatePropagation()` inside `one` would prevent `two` from being called.
+
+Duplicate attributes/properties on elements — which now includes event handlers — are not allowed. Instead, do this:
+
+```svelte
+<button
+	onclick={(e) => {
+		one(e);
+		two(e);
+	}}
+>
+	...
+</button>
+```
+
+## Why the change?
+
+By deprecating `createEventDispatcher` and the `on:` directive in favour of callback props and normal element properties, we:
+
+- reduce Svelte's learning curve
+- remove boilerplate, particularly around `createEventDispatcher`
+- remove the overhead of creating `CustomEvent` objects for events that may not even have listeners
+- add the ability to spread event handlers
+- add the ability to know which event handlers were provided to a component
+- add the ability to express whether a given event handler is required or optional (props with default values are considered optional, those without are considered required)
```

Comment:
The sentence in braces isn't completely accurate, since you could declare a handler as optional through types, and then call it as `onclick?.()`. I think it's best to just remove it
```suggestion
- add the ability to express whether a given event handler is required or optional
```